web/src/search/input/QueryInput.tsx

```diff
@@ -150,7 +151,12 @@ export class QueryInput extends React.Component<Props, State> {
         // (will be used in next PR to push to queryHistory (undo/redo))
         this.subscriptions.add(this.inputValues.subscribe(queryState => this.props.onChange(queryState)))
 
-        if (!this.props.withoutSuggestions) {
+        const hideSuggestionsSetting =
+            isSettingsValid(props.settingsCascade) &&
+            props.settingsCascade.final &&
+            props.settingsCascade.final['search.hideSuggestions']
+
+        if (!this.props.withoutSuggestions || hideSuggestionsSetting) {
```

Comment:
Unless I'm mistaken, the current check will "show suggestions on input change" if `hideSuggestionsSetting is truthy?`

```suggestion
        if (!hideSuggestionsSetting && !this.props.withoutSuggestions) {
```

Reply:
Ah yeah, sorry, my mistake!